### PR TITLE
Add session endpoint

### PR DESCRIPTION
**Pull request by InfernoJJ**

### CODE REVIEW

File: api/identityapi.yml [hunk 1]
```diff
@@ -88,6 +88,28 @@ paths:
...
+      - GatewayAuth: []
+      parameters: []
+      responses:
+        '200':
+          description: User successfully registered, they can now log in
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SessionDetails'
+        '404':
```
Comment (joshsadler) [inline api/identityapi.yml:107]: This is where I dislike 404. There are no IDs or other parameters in the path that if changed would change the response, so a 401 seems appropriate.

File: api/identityapi.yml [hunk 2]
```diff
@@ -88,6 +88,28 @@ paths:
...
+      operationId: GetSessionDetails
+      summary: Return information about the current session
+      tags:
+      - session
+      security:
+      - GatewayAuth: []
+      parameters: []
+      responses:
+        '200':
+          description: User successfully registered, they can now log in
```
Comment (adamdecaf) [inline api/identityapi.yml:102]: Doesn't this endpoint confirm their authentication is correct? Not that they've registered.

Comment (InfernoJJ) [reply api/identityapi.yml:102]: gah yeah that comment is wrong lol

File: api/identityapi.yml (hunk 1 above)
Comment (InfernoJJ) [reply api/identityapi.yml:107]: 404 is a valid response code for disallowed as not to leak the existence of it. We could change it for this one but that causes ripples into other parts of the system.

Comment (adamdecaf) [reply api/identityapi.yml:107]: Wouldn't we convert the 401 to a 404 anyway for the client?

File: api/identityapi.yml (hunk 2 above)
Comment (adamdecaf) [reply api/identityapi.yml]: Does there need to be a regeneration command ran?

File: api/identityapi.yml (hunk 1 above)
Comment (InfernoJJ) [reply api/identityapi.yml:107]: Yeah... that's why I mentioned the ripples. We'd have to have a one-off just for this endpoint in traefik to let it through.

**Pull request merged**